scripts/gen-ceph-kustomize.sh

```diff
@@ -0,0 +1,174 @@
+#!/bin/bash
+#
+# Copyright 2022 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+# set -ex
+
+# expect that the common.sh is in the same dir as the calling script
+SCRIPTPATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+. ${SCRIPTPATH}/common.sh --source-only
+
+if [ -z "$IMAGE" ]; then
+    echo "Unable to determine ceph image."
+    exit 1
+fi
+
+if [ -z "$NAMESPACE" ]; then
+    echo "Please set NAMESPACE."
+    exit 1
+fi
+
+if [ ! -d ${DEPLOY_DIR} ]; then
+    mkdir -p ${DEPLOY_DIR}
+fi
+
+pushd ${DEPLOY_DIR}
+
+POD_IP=${POD_IP:-false}
+HOSTNETWORK=${HOSTNETWORK:-false}
+NET_DISCOVERY=${NET_DISCOVERY:-"2"}
+POOLS=("volumes" "images" "backups")
+
+function add_ceph_pod {
+cat <<EOF >ceph-pod.yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: ceph
+  namespace: openstack
+  labels:
+    app.kubernetes.io/name: ceph
+    app: ceph
+spec:
+  hostNetwork: $HOSTNETWORK
+  containers:
+   - image: quay.io/ceph/daemon:latest-quincy
+     name: ceph
+     env:
+     - name: MON_IP
+       value: "$MON_IP"
+     - name: NETWORK_AUTO_DETECT
+       value: "$NET_DISCOVERY"
+     - name: CEPH_DAEMON
+       value: demo
+     - name: CEPH_PUBLIC_NETWORK
+       value: "0.0.0.0/0"
+     - name: DEMO_DAEMONS
+       value: "osd"
+     volumeMounts:
+      - mountPath: /var/lib/ceph
+        name: data
+      - mountPath: /var/log/ceph
+        name: log
+      - mountPath: /run/ceph
+        name: run
+  volumes:
+  - name: data
+    emptyDir:
+      sizeLimit: 500Mi
+  - name: run
+    emptyDir:
+      sizeLimit: 500Mi
+  - name: log
+    emptyDir:
+      sizeLimit: 500Mi
+  securityContext:
+    runAsUser: 0
+    seccompProfile:
+      type: Unconfined
+EOF
+}
+
+function ceph_kustomize {
+cat <<EOF >kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- ./ceph-pod.yaml
+namespace: openstack
+patches:
+- target:
+    kind: Pod
+  patch: |-
+    - op: replace
+      path: /metadata/namespace
+      value: $NAMESPACE
+    - op: replace
+      path: /spec/containers/0/image
+      value: $IMAGE
+EOF
+}
+
+function get_pod_ip {
+    if [[ "$HOSTNETWORK" == "false" ]]; then
+        echo "HOSTNETWORK must be set to true!"
+        exit 1
+    fi
+    MON_IP=$(oc get nodes -o 'jsonpath={.items[*].status.addresses[?(.type=="InternalIP")].address}')
+    NET_DISCOVERY="0"
+}
+
+function ceph_is_ready {
```

Comment:
Can we consider adding a timeout for this function? we could fail earlier in CI with a timeout here.
One idea is to add timeout when calling it: "timeout 300 ceph_is_ready"

Reply:
Makes sense to me, I'll rebase and add it!